packages/api/src/routers/blob/getCount.ts

```diff
@@ -19,52 +19,74 @@ export async function countBlobs(
   prisma: BlobscanPrismaClient,
   filters: Filters
 ) {
-  if (!hasCustomFilters(filters)) {
-    const overallStats = await prisma.blobOverallStats.findFirst({
-      select: {
-        totalBlobs: true,
-      },
-      where: {
-        AND: [{ category: null }, { rollup: null }],
-      },
-    });
-
-    return overallStats?.totalBlobs ?? 0;
-  }
-
   const {
+    blockNumber,
+    blockTimestamp,
     transactionAddresses,
     transactionCategory,
     transactionRollup,
     blockSlot,
     blockType,
   } = filters;
 
-  const txFiltersExists =
-    transactionRollup !== undefined ||
-    transactionAddresses ||
-    transactionCategory;
-  const blockFiltersExists = blockSlot || blockType;
+  if (hasToPerformCount(filters)) {
+    const transactionFiltersEnabled =
+      transactionCategory || transactionRollup || transactionAddresses;
+
+    return prisma.blobsOnTransactions.count({
+      where: {
+        blockNumber,
+        blockTimestamp,
+        block: {
+          slot: blockSlot,
+          transactionForks: blockType,
+        },
+        transaction: transactionFiltersEnabled
+          ? {
+              category: transactionCategory,
+              rollup: transactionRollup,
+              OR: transactionAddresses,
+            }
+          : undefined,
+      },
+    });
+  }
+
+  const { fromDay, toDay, category, rollup } = buildCountStatsFilters(filters);
 
-  return prisma.blobsOnTransactions.count({
+  if (fromDay || toDay) {
```

Comment:
A comment like this would be helpful

```suggestion
  // Counting elements in postgres can become very slow when there are millions of them in a table, so we take advantage of the calculated data used in charts
  if (fromDay || toDay) {
  ```